src/libostree/ostree-repo.c

```diff
@@ -2835,21 +2833,31 @@ reload_core_config (OstreeRepo          *self,
         if (!min_free_space_size_validate_and_convert (self, min_free_space_size_str, error))
           return glnx_prefix_error (error, "Invalid min-free-space-size '%s'", min_free_space_size_str);
       }
-    else
+
+    if (g_key_file_has_key (self->config, "core", "min-free-space-percent", NULL))
       {
         g_autofree char *min_free_space_percent_str = NULL;
-        /* If changing this, be sure to change the man page too */
-        const char *default_min_free_space = "3";
 
         if (!ot_keyfile_get_value_with_default (self->config, "core", "min-free-space-percent",
-                                                default_min_free_space,
-                                                &min_free_space_percent_str, error))
+                                                NULL, &min_free_space_percent_str, error))
           return FALSE;
 
         self->min_free_space_percent = g_ascii_strtoull (min_free_space_percent_str, NULL, 10);
         if (self->min_free_space_percent > 99)
           return glnx_throw (error, "Invalid min-free-space-percent '%s'", min_free_space_percent_str);
       }
+    else if (!g_key_file_has_key (self->config, "core", "min-free-space-size", NULL))
+      {
+        /* Default fallback of 3% free space. If changing this, be sure to change the man page too */
+        self->min_free_space_percent = 3;
+        self->min_free_space_mb = 0;
+      }
+
+    if (self->min_free_space_percent != 0 && self->min_free_space_mb != 0)
+      {
+        self->min_free_space_percent = 0;
+        g_warning ("Both min-free-space-percent and -size are mentioned in config. Enforcing min-free-space-size check only.");
```

Comment:
I am not a fan of `g_warning()` in general.  It seriously annoys me how prevalent it is in GNOME - there's a constant stream of warnings that we've shipped to users forever from apps and the core that end up spamming the journal and don't really get fixed.

It's pretty similar to what ends up happening with Python apps when a library deprecates an API - those aren't for users to fix, they're for developers.

rpm-ostree's test suite runs with `G_DEBUG=fatal-warnings` and I am tempted to turn that on by default too.

Since we *know* people are going to hit this for a while, let's just not emit a warning.  At some point the newer ostree and flatpak will propagate most places and there won't be a problem anymore.

I am not too concerned about the case of an admin adding both manually.  If they don't read the docs that's not our fault.

Reply:
(Sort of tangentially related...an interesting middle ground is what Linux has with e.g. `WARN_ON_ONCE()` - except that's hard to implement in userspace since it's very common for processes to be restarted and maintaining the warning state outside would get tricky - would probably require assistance from journald somehow)